store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -1208,7 +1208,12 @@ private void doCommit() {
                         flushOK = CommitLog.this.mappedFileQueue.getFlushedWhere() >= req.getNextOffset();
                     }
 
-                    req.wakeupCustomer(flushOK ? PutMessageStatus.PUT_OK : PutMessageStatus.FLUSH_DISK_TIMEOUT);
+                    if (CommitLog.this.mappedFileQueue.isFlushError()) {
+                        req.wakeupCustomer(PutMessageStatus.FLUSH_DISK_FAILED);
+                    }
+                    else {
+                        req.wakeupCustomer(flushOK ? PutMessageStatus.PUT_OK : PutMessageStatus.FLUSH_DISK_TIMEOUT);
+                    }
                 }
```

Comment:
May be it's better to format:
if () {
  xxx
} else {
  xxx
}

Reply:
@hzh0425 thanks.changed.